App.tsx

```diff
@@ -1,28 +1,57 @@
+import AsyncStorage from '@react-native-community/async-storage'
 import React, { Component } from 'react'
-import { StatusBar } from 'react-native'
+import { Image, StatusBar, View } from 'react-native'
 import { Provider } from 'react-redux'
+import { setGlobal } from 'reactn'
+import { PV } from './src/resources'
 import Router from './src/Router'
 import { configureStore } from './src/store/store'
+import { colors, darkTheme, lightTheme } from './src/styles'
 
 type Props = {}
 
-type State = {}
+type State = {
+  appReady: boolean
+}
 
 interface App {
   store?: any
 }
 
 class App extends Component<Props, State> {
-  constructor (props: Props) {
+  constructor(props: Props) {
     super(props)
     this.store = configureStore()
     StatusBar.setBarStyle('light-content')
+    this.state = {
+      appReady: false
+    }
+  }
+
+  async componentDidMount() {
+    const darkModeEnabled = await AsyncStorage.getItem('DARK_MODE_ENABLED')
+    this.setupGlobalState(darkModeEnabled ? darkTheme : lightTheme)
+    this.setState({ appReady: true })
+  }
+
+  setupGlobalState(theme: {}) {
```

Comment:
What does `theme: {}` do here? Set the theme parameter as an object type?

Reply:
Here it just indicates that theme is of object type. I should have probably just set it as "object" but "{}" worked too. Feel free to change it if you prefer